server/src/main/java/org/elasticsearch/index/mapper/InferenceMetadataFieldsMapper.java

```diff
@@ -35,7 +36,12 @@ public abstract class InferenceMetadataFieldsMapper extends MetadataFieldMapper
      */
     public static final Setting<Boolean> USE_LEGACY_SEMANTIC_TEXT_FORMAT = Setting.boolSetting(
         "index.mapping.semantic_text.use_legacy_format",
-        false,
+        s -> {
+            // Check index version SOURCE_MAPPER_MODE_ATTRIBUTE_NOOP because that index version was added in the same serverless promotion
+            // where the new format was enabled by default
+            IndexVersion indexVersion = IndexMetadata.SETTING_INDEX_VERSION_CREATED.get(s);
+            return Boolean.toString(indexVersion.before(IndexVersions.SOURCE_MAPPER_MODE_ATTRIBUTE_NOOP));
```

Comment:
Nitpick - put the index version in a variable and the comment there, then the code can just be something like `return Boolean.toString(indexVersion.before(NEW_SEMANTIC_TEXT_FORMAT_RELEASED)`